mapie/conformity_scores/residual_conformity_scores.py

```diff
@@ -132,3 +141,275 @@ def get_estimation_distribution(
+
+        Parameters
+        ----------
+        X : NDArray
+            All the observed values used in the general fit.
+
+        y : NDArray
+            All the observed targets used in the general fit.
+
+        y_pred : NDArray
+            Predicted targets.
+
+        calres_indexes : NDArray
+            Indexes used for the training of the estimaot and the calibration.
+
+        Returns
+        -------
+        Tuple[NDArray, NDArray]
+            - indexes needed for the calibration.
+            - indexes used for the training of the base estimator.
+        """
+        (X_res_indexes,
+         X_cal_indexes,
+         y_res_indexes,
+         y_cal_indexes) = train_test_split(
+            calres_indexes,
+            calres_indexes,
+            test_size=self.split_size,
+            random_state=self.random_state,
+        )
+
+        residuals = np.abs(np.subtract(
+            y[y_res_indexes],
+            y_pred[y_res_indexes]
+        ))
+        residual_estimator_targets = np.log(np.maximum(
+            residuals,
+            np.full(residuals.shape, self.eps)
+        ))
+
+        residual_estimator_ = residual_estimator_.fit(
+            X[X_res_indexes],
+            residual_estimator_targets
+        )
+
+        cal_index = X_cal_indexes
+        train_index = list(set(np.arange(y_pred.shape[0])) - set(cal_index))
+
+        self.residual_estimator_ = residual_estimator_
+
+        return cal_index, np.array(train_index)
+
+    def get_signed_conformity_scores(
+        self,
+        X: ArrayLike,
+        y: ArrayLike,
+        y_pred: ArrayLike
+    ) -> NDArray:
+        """
+        Computes the signed conformity score = (y - y_pred) / r_pred.
+        r_pred being the predicted residual (y - y_pred) of the estimator.
+        It is calculated by a model (``residual_estimator_``) that learns
+        to predict this residual.
+
+        The learning is done with the log of the residual and later we
+        use the exponential of the prediction to avoid negative values.
+        """
+        X, y, y_pred, self.residual_estimator_ = self._check_parameters(
+            X, y, y_pred
+        )
+        calres_indexes = np.argwhere(
+                            np.logical_not(np.isnan(y_pred))
+                        ).reshape((-1, ))
+        if not self.prefit:
+            cal_indexes, train_indexes = self._fit_residual_estimator(
+                clone(self.residual_estimator_), X, y, y_pred, calres_indexes
+            )
+        else:
+            cal_indexes = calres_indexes
+            train_indexes = np.argwhere(np.isnan(y_pred)).reshape((-1,))
+
+        normalizer = np.maximum(
+            np.exp(self.residual_estimator_.predict(X[cal_indexes])),
+            self.eps
+        )
+        signed_conformity_scores = np.divide(
+            np.abs(np.subtract(y[cal_indexes], y_pred[cal_indexes])),
+            normalizer
+        )
```

Comment:
```suggestion
        residuals_pred = np.maximum(
            np.exp(self.residual_estimator_.predict(X[cal_indexes])),
            self.eps
        )
        signed_conformity_scores = np.divide(
            np.subtract(y[cal_indexes], y_pred[cal_indexes]),
            residuals_pred
        )
```

Reply:
Done (without the suppression of np.abs as I think it is a typo)

Reply:
Your variable is called `signed_conformity_scores`. So, I think by keeping `np.abs`, you compute `unsigned_conformity_scores`. Did I misunderstand something here?

Reply:
Done